Raven.Database/Bundles/ScriptedIndexResults/ScriptedIndexResultsIndexTrigger.cs

```diff
@@ -71,7 +72,7 @@ public override void OnIndexEntryCreated(string entryKey, Document document)
             {
                 if (created.ContainsKey(entryKey) == false)
```

Comment:
hi @efratshenhar 
you probably should use 

var bag = created.GetOrAdd(entryKey, () => new ConcurrentBag<RavenJObject>());
bag.Add(CreateJsonDocumentFromLuceneDocument(document));

for this code to work as expected. Between ContainsKey and created[entryKey] a lot can happen in multi-threaded environment and you want this to be an atomic operation.

Also, could you reset your changes to the ayende/master and apply them once again. In your PR there are 9 commits.